### PR TITLE
adds chatml prompt template as string to maintain configurability

### PR DESCRIPTION
**Pull request by biswaroop1547**
and has a generic stitch_prompt function which can be reused for other services for prompt template configurability.

Fixes https://github.com/premAI-io/prem-app/issues/401

### CODE REVIEW

**Comment by biswaroop1547** [comment]
relevant: https://github.com/abetlen/llama-cpp-python/issues/717

**Comment by casperdcl** [comment]
fixes https://github.com/premAI-io/prem-app/issues/401

**Review by filopedraz — DISMISSED** [review]

File: cht-llama-cpp/models.py [hunk 1]
```diff
@@ -43,22 +57,58 @@ def generate(
...
             stream=stream,
             stop=stop,
             max_tokens=max_tokens,
         )
+        if not stream and cls.PROMPT_TEMPLATE.get("template_format") == "chatml":
+            cht_resp["choices"][0]["message"]["content"] = (
+                cht_resp["choices"][0]["message"]["content"].split("\n<|im_end|>")[0].strip()
+            )
+
+        # TODO: handle postprocessing for streaming responses
```
Comment (casperdcl) [inline cht-llama-cpp/models.py:73]: In this PR or a follow-up?

File: cht-llama-cpp/models.py [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
 
 DEFAULT_N_THREADS = max(multiprocessing.cpu_count() // 2, 1)
 
 
+@llama_chat_format.register_chat_format("chatml")
+def initiate_chatml_prompt_template(
+    messages: List[llama_types.ChatCompletionRequestMessage],
+    **kwargs: Any,
+) -> llama_chat_format.ChatFormatterResponse:
+    # Until https://github.com/abetlen/llama-cpp-python/issues/717 supports ChatML.
```
Comment (casperdcl) [inline cht-llama-cpp/models.py:15]:
```suggestion
    # TODO: drop when https://github.com/abetlen/llama-cpp-python/issues/717 supports ChatML
```

Easier to find `TODO` strings

File: cht-llama-cpp/models.py (hunk 1 above)
Comment (biswaroop1547) [reply cht-llama-cpp/models.py:73]: will be a follow-up PR (mostly later)

**Pull request merged**